pytest_doctestplus/plugin.py

```diff
@@ -187,28 +187,35 @@ class DocTestModulePlus(doctest_plugin.DoctestModule):
         # behavior (which doesn't do whitespace normalization or
         # handling __doctest_skip__) doesn't happen.
         def collect(self):
+            if PYTEST_GE_6_3:
+                path = self.path
+            else:
+                path = Path(self.fspath)
+
             # When running directly from pytest we need to make sure that we
             # don't accidentally import setup.py!
-            if self.fspath.basename == "setup.py":
+            if path.name == "setup.py":
                 return
-            elif self.fspath.basename == "conftest.py":
+            elif path.name == "conftest.py":
                 if PYTEST_GE_6_3:
                     module = self.config.pluginmanager._importconftest(
-                        Path(self.fspath), self.config.getoption("importmode"))
+                        path, self.config.getoption("importmode"))
                 elif PYTEST_GT_5:
                     module = self.config.pluginmanager._importconftest(
                         self.fspath, self.config.getoption("importmode"))
                 else:
                     module = self.config.pluginmanager._importconftest(
                         self.fspath)
-            else:
+            elif not PYTEST_GE_6_3:
                 try:
                     module = self.fspath.pyimport()
                 except ImportError:
                     if self.config.getvalue("doctest_ignore_import_errors"):
                         pytest.skip("unable to import module %r" % self.fspath)
                     else:
                         raise
+            else:
+                raise NotImplementedError("unable to parse %r" % path)
```

Comment:
I don't know how to replace `self.fspath.pyimport()` here, as `pytest` is dropping support for `py.path.local`.

Reply:
Hmm, so if `self.path` is a pathlib object I guess we need to import to module with `importlib.import_module` ?

Reply:
Yes, I think `self.path` is `pathlib.Path`. Should we ping a pytest dev for review?

Reply:
Maybe useful to know if they have a better solution or if we should re-implement the part of `.pyimport` that we need (https://github.com/pytest-dev/py/blob/d4111cae5572809a23d3d5cc88bed8e4af720929/py/_path/local.py#L649-L650)

Reply:
@nicoddemus perhaps or at least he'll know who else to ping? 🙏

Reply:
Hey!

An alternative is to import the internal [import_path() function we use](https://github.com/pytest-dev/pytest/blob/bfbf733a3ed2f58a71570aa6e950444f3c5b0ea1/src/_pytest/pathlib.py#L456), however this is not ideal in the long term.

That function is a bit long because of the different import modes we support, however if you would like to only support the newer `importlib` mode, the code is relatively small and uses only the `importlib` stable API.

https://github.com/pytest-dev/pytest/blob/bfbf733a3ed2f58a71570aa6e950444f3c5b0ea1/src/_pytest/pathlib.py#L489-L504

cc @RonnyPfannschmidt @asottile in case they have other suggestions.